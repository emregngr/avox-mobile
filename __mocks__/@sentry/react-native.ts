import type { ComponentType } from 'react'

const mockedSentry = {
  init: jest.fn(),
  wrap: <T extends ComponentType<any>>(Component: T): T => Component,
  withScope: jest.fn((callback: (scope: any) => void) =>
    callback({
      setTag: jest.fn(),
      setLevel: jest.fn(),
      setUser: jest.fn(),
      setContext: jest.fn(),
    }),
  ),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  addBreadcrumb: jest.fn(),
  configureScope: jest.fn(),
  setUser: jest.fn(),
  setTag: jest.fn(),
  setContext: jest.fn(),
  setRelease: jest.fn(),
  setDist: jest.fn(),
  setEnvironment: jest.fn(),
  close: jest.fn(() => Promise.resolve(true)),
  flush: jest.fn(() => Promise.resolve(true)),
}

module.exports = mockedSentry

export type mockedSentryType = typeof mockedSentry
