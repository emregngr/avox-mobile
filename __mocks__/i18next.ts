const mockedInit = jest.fn().mockResolvedValue(undefined)
const mockedUse = jest.fn().mockReturnValue({ init: mockedInit })

export default {
  language: 'en',
  use: mockedUse,
  init: mockedInit,
}
