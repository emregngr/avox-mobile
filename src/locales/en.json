{
  "connectionError": "Connection Error",
  "connectionErrorDescription": "Please check your internet connection and try again.",
  "ok": "OK",
  "minPasswordMessage": "Password must be at least 6 characters.",
  "passwordsDoNotMatch": "Passwords do not match.",
  "newPasswordCannotBeSame": "New password cannot be the same as the current password.",
  "currentPassword": "Current Password",
  "currentPasswordPlaceholder": "Enter your current password",
  "newPassword": "New Password",
  "newPasswordPlaceholder": "Enter your new password",
  "confirmNewPassword": "Confirm New Password",
  "confirmNewPasswordPlaceholder": "Re-enter your new password",
  "changePassword": "Change Password",
  "addPassword": "Add Password",
  "error": "Error",
  "chooseLanguage": "Choose Language",
  "light": "Light",
  "dark": "Dark",
  "chooseTheme": "Choose Theme",
  "modeSelection": "Mode Selection",
  "minFirstNameMessage": "First name must be at least 2 characters.",
  "minLastNameMessage": "Last name must be at least 2 characters.",
  "invalidEmailMessage": "Invalid email address.",
  "information": "Information",
  "havenNotMadeChanges": "You have not made any changes.",
  "updateProfile": "Update Profile",
  "firstNamePlaceholder": "Enter your first name",
  "firstName": "First Name",
  "lastNamePlaceholder": "Enter your last name",
  "lastName": "Last Name",
  "emailPlaceholder": "Enter your email address",
  "email": "Email",
  "update": "Update",
  "avox": "Avox",
  "welcomeText": "Welcome",
  "signInWithGoogle": "Sign in with Google",
  "signInWithApple": "Sign in with Apple",
  "or": "or",
  "login": "Login",
  "register": "Register",
  "acceptText": "I have read, understood, and agree to the Privacy Policy and Terms of Use.",
  "emailSent": "Email Sent",
  "passwordResetLinkSentMessage": "Password reset link has been sent to your email address.",
  "somethingWentWrong": "Something went wrong.",
  "forgotPassword": "Forgot Password",
  "sendResetLink": "Send Reset Link",
  "maxPasswordMessage": "Password cannot exceed 20 characters.",
  "password": "Password",
  "passwordPlaceholder": "Enter your password",
  "dontHaveAccount": "Don't have an account?",
  "createAccount": "Create Account",
  "privacyPolicy": "Privacy Policy",
  "accept": "Accept",
  "privacyPolicyMessage": "You must accept the Privacy Policy.",
  "termsOfUseMessage": "You must accept the Terms of Use.",
  "termsOfUse": "Terms of Use",
  "alreadyHaveAccount": "Already have an account?",
  "breakingNewDetailTitle": "News Detail",
  "forceUpdateText": "A new version of the app is available. Please update.",
  "forceUpdateButton": "Update",
  "maintenanceText": "System maintenance in progress. Please try again later.",
  "onBoardingTitle1": "Welcome to Avox",
  "onBoardingTitle2": "Explore Airlines",
  "onBoardingTitle3": "Discover Airports",
  "onBoardingTitle4": "Simple, Fast and Smart",
  "onBoardingText1": "Everything about aviation is now at your fingertips.",
  "onBoardingText2": "Access fleets, routes, and more with just a tap.",
  "onBoardingText3": "Airports around the world, ready for you to explore.",
  "onBoardingText4": "A modern design that makes finding information effortless.",
  "faq": "FAQ",
  "ratingRequestNotSupported": "Rating not supported on this platform.",
  "deleteAccount": "Delete Account",
  "deleteAccountConfirmation": "Are you sure you want to delete your account? This action cannot be undone.",
  "cancel": "Cancel",
  "delete": "Delete",
  "settings": "Settings",
  "general": "General",
  "notification": "Notification",
  "rateApp": "Rate App",
  "socialMediaAccounts": "Social Media Accounts",
  "tikTok": "TikTok",
  "instagram": "Instagram",
  "profileInformation": "Profile Information",
  "accountTransactions": "Account Transactions",
  "warning": "Warning",
  "logoutWarning": "Are you sure you want to log out?",
  "yes": "Yes",
  "logout": "Logout",
  "home": "Home",
  "discover": "Discover",
  "favorites": "Favorites",
  "profile": "Profile",
  "europe": "Europe",
  "asia": "Asia",
  "northAmerica": "North America",
  "southAmerica": "South America",
  "africa": "Africa",
  "oceania": "Oceania",
  "numberOfDestination": "Destinations",
  "numberOfCountries": "Countries",
  "mega": "Mega",
  "large": "Large",
  "medium": "Medium",
  "small": "Small",
  "open24Hours": "Open 24 Hours",
  "luggage": "Luggage",
  "terminal": "Terminal",
  "terminalArea": "Terminal Area (Ha)",
  "airportArea": "Area (Ha)",
  "apron": "Apron",
  "dutyFree": "Duty Free",
  "restaurants": "Restaurants",
  "lounges": "Lounges",
  "freeWifi": "Free WiFi",
  "carRental": "Car Rental",
  "healthServices": "Health Services",
  "childrensPlayground": "Children's Playground",
  "placeOfWorship": "Place of Worship",
  "hotels": "Hotels",
  "runway": "Runway",
  "runwaym": "Runway (m)",
  "tower": "Control Tower (m)",
  "parking": "Parking",
  "metro": "Metro",
  "security": "Security (min)",
  "checkIn": "Check-in (min)",
  "googleRating": "Google Rating",
  "majorInternational": "Major International",
  "regional": "Regional",
  "lowCost": "Low-Cost",
  "cargo": "Cargo",
  "averageAgeYears": "Average Age",
  "airplaneType": "Airplane Type",
  "none": "None",
  "skytraxRating": "Skytrax Rating",
  "clear": "Clear",
  "route": "Route",
  "country": "Country",
  "rating": "Rating",
  "popularRoutes": "Popular Routes",
  "noAirline": "No airline found.",
  "allAirlinesShown": "All airlines are shown.",
  "iata": "IATA",
  "icao": "ICAO",
  "continent": "CONTINENT",
  "totalDestination": "Total Destination",
  "domesticDestinations": "Domestic Destinations",
  "internationalDestination": "International Destinations",
  "companyInformation": "Company Information",
  "yearOfEstablishment": "Year",
  "parentCompany": "Parent Company",
  "numberOfPassengers": "Passengers",
  "numberOfEmployees": "Employees",
  "businessModel": "Business Model",
  "businessType": "Business Type",
  "isoCountryCode": "ISO Country Code",
  "isoRegionCode": "ISO Region Code",
  "contactInformation": "Contact Information",
  "website": "Website",
  "phone": "Phone",
  "socialMedia": "Social Media",
  "hubInformation": "Hub Information",
  "city": "City",
  "hubLocation": "Hub Location",
  "totalAirplane": "Total Aircraft",
  "count": "Count",
  "kmh": "km/h",
  "km": "km",
  "person": "Person",
  "ton": "ton",
  "safetyInformation": "Safety Information",
  "safetyRecord": "Safety Record",
  "certifications": "Certifications",
  "environmentalPolicy": "Environmental Policy",
  "environmentalResponsibility": "Environmental Responsibility",
  "services": "Services",
  "noAirports": "No airports found.",
  "allAirportsShown": "All airports are shown.",
  "region": "Region",
  "year": "Year",
  "million": "Million",
  "airportType": "Airport Type",
  "scheduledService": "Scheduled Service",
  "no": "No",
  "locationInformation": "Location Information",
  "elevation": "Elevation (ft)",
  "airportLocation": "Airport Location",
  "terminalandCapacity": "Terminal and Capacity",
  "million/year": "million/year",
  "baggageCapacity": "Baggage Capacity",
  "luggage/hour": "baggage/hour",
  "trackInformation": "Runway Information",
  "surface": "Surface",
  "pcn": "PCN",
  "ilsCategory": "ILS Category",
  "facilities": "Facilities",
  "fireDepartmentCategory": "Fire Department Category",
  "wifiStatus": "WiFi Status",
  "free": "Free",
  "paid": "Paid",
  "annualShipping": "Annual Shipping (ton)",
  "terminalCapacity": "Terminal Capacity (ton)",
  "existing": "Existing",
  "coldStorage": "Cold Storage",
  "dangerousGoods": "Dangerous Goods",
  "getDirection": "Get Directions",
  "filter": "Filter",
  "additionalServicesFacilities": "Additional Services and Facilities",
  "alliance": "Alliance",
  "minGoogleRating": "Minimum Google Rating",
  "minSkytraxRating": "Minimum Skytrax Rating",
  "apply": "Apply",
  "flight/year": "Flights/Year",
  "distance": "Distance",
  "viewAll": "View All",
  "successful": "Successful",
  "profileUpdatedSuccessfully": "Profile updated successfully.",
  "profileUpdateFailed": "Profile update failed.",
  "passwordChangedSuccessfully": "Password changed successfully.",
  "passwordAddedSuccessfully": "Password added successfully.",
  "accountDeletedSuccessfully": "Account deleted successfully.",
  "accountDeleteFailed": "Account deletion failed.",
  "googleSignInFailed": "Google sign-in failed.",
  "appleSignInFailed": "Apple sign-in failed.",
  "invalidEmail": "Invalid email address.",
  "userNotLoggedIn": "User not logged in.",
  "add": "Add",
  "airlineSearchPlaceholder": "Airline, country, IATA code search...",
  "airportSearchPlaceholder": "Airport, country, IATA code search...",
  "airports": "Airports",
  "airlines": "Airlines",
  "company": "Company",
  "fleet": "Fleet",
  "network": "Network",
  "safety": "Safety",
  "infrastructure": "Infra",
  "flights": "Flights",
  "airplane": "Airplane",
  "minutes": "minutes",
  "passenger": "Passenger",
  "noFavoriteAirport": "No Favorite Airport",
  "noFavoriteAirline": "No Favorite Airline",
  "mainDestinations": "Main Destinations",
  "activeRoutes": "Active Routes",
  "fleetDetail": "Fleet Detail",
  "news": "News",
  "popularAirlines": "Popular Airlines",
  "popularAirports": "Popular Airports",
  "popularDestinations": "Popular Destinations",
  "totalAirplanes": "Total Airplanes",
  "ctry": "Ctry",
  "skip": "Skip",
  "continue": "Continue",
  "passwordChangeFailed": "Password change failed.",
  "passwordAddFailed": "Password add failed.",
  "cont": "Cont",
  "reg": "Reg",
  "appleMaps": "Open in Apple Maps",
  "googleMaps": "Open in Google Maps",
  "selectMapApp": "Select Map Application",
  "signInOrRegister": "Sign In / Register",
  "loading": "Loading...",
  "passwordResetLinkSentMessageFailed": "The password reset link could not be sent to your email address.",
  "logoutFailed": "Logout failed.",
  "socialLoginFailed": "Social login failed.",
  "emailLoginFailed": "Email or password incorrect.",
  "emailRegisterFailed": "Registration by email failed.",
  "airlineInfo": "Airline Information",
  "airportInfo": "Airport Information",
  "retry": "Retry",
  "change": "Change"
}
