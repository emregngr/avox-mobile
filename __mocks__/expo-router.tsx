import { createElement, Fragment, ReactNode } from 'react'
import { View } from 'react-native'

const mockedRouterFunctions = {
  push: jest.fn(),
  navigate: jest.fn(),
  replace: jest.fn(),
  back: jest.fn(),
  canGoBack: jest.fn(() => true),
  setParams: jest.fn(),
}

let mockedPathname = '/home'

const setMockPathname = (pathname: string) => {
  mockedPathname = pathname
  const usePathnameMock = module.exports.usePathname
  if (usePathnameMock && usePathnameMock.mockReturnValue) {
    usePathnameMock.mockReturnValue(mockedPathname)
  }
}

const clearAllMocks = () => {
  Object.values(mockedRouterFunctions).forEach(fn => {
    if (typeof fn === 'function' && fn.mockClear) {
      fn.mockClear()
    }
  })
}

const MockScreen = ({ children }: { children: ReactNode }) =>
  createElement(Fragment, null, children)

const MockedStack = ({ children }: { children: ReactNode }) =>
  createElement(Fragment, null, children)

const MockedTabs = ({
  children,
  screenOptions,
  ...props
}: {
  children: ReactNode
  screenOptions?: any
  [key: string]: any
}) => (
  <View testID="tabs" data-screen-options={JSON.stringify(screenOptions)} {...props}>
    {children}
  </View>
)

const MockTabScreen = ({ name, ...props }: { name: string; [key: string]: any }) => (
  <View testID={`tab-screen-${name}`} data-name={name} {...props} />
)

module.exports = {
  router: mockedRouterFunctions,
  useRouter: jest.fn(() => mockedRouterFunctions),
  useFocusEffect: jest.fn(),
  useSegments: jest.fn(() => []),
  useSearchParams: jest.fn(() => ({})),
  useLocalSearchParams: jest.fn(() => ({})),
  useGlobalSearchParams: jest.fn(() => ({})),
  usePathname: jest.fn(() => mockedPathname),
  useNavigation: jest.fn(() => ({})),
  Slot: ({ children }: { children: ReactNode }) => <>{children}</>,
  Stack: Object.assign(MockedStack, { Screen: MockScreen }),
  Tabs: Object.assign(MockedTabs, { Screen: MockTabScreen }),
  Link: ({ children, ...props }: { children: ReactNode }) =>
    createElement(Fragment, null, children),
  Redirect: () => null,
  setMockPathname: setMockPathname,
  clearAllMocks: clearAllMocks,
  mockedFunctions: mockedRouterFunctions,
  get mockPathname() {
    return mockedPathname
  },
}
