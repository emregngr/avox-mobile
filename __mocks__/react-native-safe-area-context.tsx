import { ComponentType, ReactNode } from 'react'
import { View } from 'react-native'

export type Insets = {
  top: number
  bottom: number
  left: number
  right: number
}

export type Frame = {
  x: number
  y: number
  width: number
  height: number
}

type SafeAreaConsumerProps = {
  children: (insets: Insets) => ReactNode
}

export const SafeAreaProvider = ({ children }: { children: ReactNode }) => <>{children}</>

export const SafeAreaConsumer = ({ children }: SafeAreaConsumerProps) =>
  children({ top: 0, bottom: 0, left: 0, right: 0 })

export const SafeAreaView = View

export const useSafeAreaInsets = (): Insets => ({
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
})

export const mockedUseSafeAreaInsets = jest.fn().mockReturnValue({
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
})

export const useSafeAreaFrame = (): Frame => ({
  x: 0,
  y: 0,
  width: 375,
  height: 812,
})

export function withSafeAreaInsets<P extends object>(
  Component: ComponentType<P>,
): ComponentType<P> {
  return Component
}
