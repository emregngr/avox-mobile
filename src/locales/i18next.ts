import { getLocales } from 'expo-localization'
import i18next from 'i18next'
import { getI18n, initReactI18next } from 'react-i18next'

import en from '@/locales/en.json'
import tr from '@/locales/tr.json'

const locales = getLocales()
i18next.language = locales[0]?.languageTag ?? 'en'

export const languageResources = i18next.use(initReactI18next).init({
  fallbackLng: 'en',
  resources: {
    en: { translation: en },
    tr: { translation: tr },
  },
})

export const getLocale = (key: string, params?: Record<string, any>) => {
  const i18n = getI18n()
  let translatedText = i18n?.t(key)

  if (params) {
    Object.keys(params).forEach(paramKey => {
      translatedText = translatedText.replace(`{${paramKey}}`, params[paramKey])
    })
  }

  return translatedText
}

export const i18nChangeLocale = async (lng: string) => {
  const i18n = getI18n()
  await i18n?.changeLanguage(lng)
}

export default i18next
