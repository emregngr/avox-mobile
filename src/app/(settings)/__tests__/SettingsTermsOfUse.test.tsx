import { fireEvent, render } from '@testing-library/react-native'
import { router } from 'expo-router'
import type { ReactNode } from 'react'
import React from 'react'
import { SafeAreaProvider } from 'react-native-safe-area-context'

import SettingsTermsOfUse from '@/app/(settings)/settings-terms-of-use'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'

jest.mock('@/locales/i18next')

const mockedGetLocale = getLocale as jest.MockedFunction<typeof getLocale>

jest.mock('@/store/locale')

const mockedUseLocaleStore = useLocaleStore as jest.MockedFunction<typeof useLocaleStore>

jest.mock('@/store/theme')

const mockedUseThemeStore = useThemeStore as jest.MockedFunction<typeof useThemeStore>

jest.mock('@/components/common', () => {
  const { View, Text, TouchableOpacity } = require('react-native')

  return {
    Header: ({ title, backIconOnPress }: { title: string; backIconOnPress: () => void }) => (
      <>
        <Text testID="header-title">{title}</Text>
        <TouchableOpacity onPress={backIconOnPress} testID="back-button">
          <Text>Back</Text>
        </TouchableOpacity>
      </>
    ),

    SafeLayout: ({ children }: { children: ReactNode }) => (
      <View testID="safe-layout">{children}</View>
    ),

    ThemedText: ({
      children,
      testID,
      color,
      type,
    }: {
      children: ReactNode
      testID?: string
      color?: string
      type?: string
    }) => <Text testID={testID || `themed-text-${type}-${color}`}>{children}</Text>,
  }
})

const renderWithSafeAreaProvider = (component: ReactNode) =>
  render(
    <SafeAreaProvider
      initialMetrics={{
        insets: { top: 0, left: 0, right: 0, bottom: 0 },
        frame: { x: 0, y: 0, width: 375, height: 812 },
      }}
    >
      {component}
    </SafeAreaProvider>,
  )

beforeEach(() => {
  mockedUseLocaleStore.mockReturnValue({
    selectedLocale: 'en',
  })

  mockedUseThemeStore.mockReturnValue({
    selectedTheme: 'light',
  })

  mockedGetLocale.mockImplementation((key: string) => {
    const translations: Record<string, string> = {
      termsOfUse: 'Terms of Use',
    }
    return translations[key] || key
  })
})

describe('SettingsTermsOfUse Screen', () => {
  it('should render terms of use screen with header and initial state', () => {
    const { getByTestId, getByText } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    expect(getByTestId('safe-layout')).toBeTruthy()
    expect(getByTestId('header-title')).toBeTruthy()
    expect(getByText('Terms of Use')).toBeTruthy()
  })

  it('should call router.back when back button is pressed', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const backButton = getByTestId('back-button')
    fireEvent.press(backButton)

    expect(router.back).toHaveBeenCalledTimes(1)
  })

  it('should render English terms of use text when locale is en', () => {
    mockedUseLocaleStore.mockReturnValue({
      selectedLocale: 'en',
    })

    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Terms of Use')
    expect(themedText.props.children).toContain(
      'By using our application, you agree to the following terms',
    )
    expect(themedText.props.children).toContain('Terms of Service')
    expect(themedText.props.children).toContain('User Account and Security')
  })

  it('should render Turkish terms of use text when locale is tr', () => {
    mockedUseLocaleStore.mockReturnValue({
      selectedLocale: 'tr',
    })

    mockedGetLocale.mockImplementation((key: string) => {
      const translations: Record<string, string> = {
        termsOfUse: 'Kullanım Koşulları',
      }
      return translations[key] || key
    })

    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Kullanım Koşulları')
    expect(themedText.props.children).toContain(
      'Uygulamamızı kullanarak aşağıdaki şartları kabul etmiş sayılırsınız',
    )
    expect(themedText.props.children).toContain('Hizmet Kullanım Koşulları')
    expect(themedText.props.children).toContain('Kullanıcı Hesabı ve Güvenlik')
  })

  it('should display user responsibility information', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('responsible for all activities')
    expect(themedText.props.children).toContain('protecting the confidentiality')
  })

  it('should display prohibited uses section', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Prohibited Uses')
    expect(themedText.props.children).toContain('illegal, harmful, or inappropriate content')
    expect(themedText.props.children).toContain('harassing other users')
    expect(themedText.props.children).toContain('commercial purposes')
  })

  it('should display intellectual property rights information', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Intellectual Property Rights')
    expect(themedText.props.children).toContain('intellectual property of our company')
    expect(themedText.props.children).toContain('prohibited to copy, modify, or distribute')
  })

  it('should display service changes information', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Service Changes')
    expect(themedText.props.children).toContain('reserve the right to modify')
    expect(themedText.props.children).toContain('suspend, or terminate')
  })

  it('should display limitation of liability section', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Limitation of Liability')
    expect(themedText.props.children).toContain('as is')
    expect(themedText.props.children).toContain('uninterrupted or error-free')
    expect(themedText.props.children).toContain('not liable for any damages')
  })

  it('should display account termination information', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Account Termination')
    expect(themedText.props.children).toContain('suspend or terminate accounts')
    expect(themedText.props.children).toContain('terminate your account at any time')
  })

  it('should display contact and support information', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Contact and Support')
    expect(themedText.props.children).toContain('support team')
    expect(themedText.props.children).toContain('assist you as soon as possible')
  })

  it('should display updates and changes information', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Updates and Changes')
    expect(themedText.props.children).toContain('update these terms')
    expect(themedText.props.children).toContain('continued use of the application')
  })

  it('should display applicable law information', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Applicable Law')
    expect(themedText.props.children).toContain('Republic of Turkey')
    expect(themedText.props.children).toContain('courts of Turkey')
  })

  it('should display Turkish legal information when locale is tr', () => {
    mockedUseLocaleStore.mockReturnValue({
      selectedLocale: 'tr',
    })

    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Türkiye Cumhuriyeti yasalarına')
    expect(themedText.props.children).toContain('Türkiye mahkemeleri')
  })

  it('should display privacy and data use information', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('Privacy and Data Use')
    expect(themedText.props.children).toContain('Privacy Policy')
    expect(themedText.props.children).toContain('consent to the collection and use')
  })

  it('should display user content responsibility', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    const themedText = getByTestId('themed-text-body1-text-100')
    expect(themedText.props.children).toContain('User Content')
    expect(themedText.props.children).toContain('responsible for all content you upload')
    expect(themedText.props.children).toContain('not infringe on copyrights')
  })

  it('should work with dark theme', () => {
    mockedUseThemeStore.mockReturnValue({
      selectedTheme: 'dark',
    })

    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    expect(getByTestId('safe-layout')).toBeTruthy()
    expect(getByTestId('themed-text-body1-text-100')).toBeTruthy()
  })

  it('should work with light theme', () => {
    mockedUseThemeStore.mockReturnValue({
      selectedTheme: 'light',
    })

    const { getByTestId } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    expect(getByTestId('safe-layout')).toBeTruthy()
    expect(getByTestId('themed-text-body1-text-100')).toBeTruthy()
  })
})

describe('SettingsTermsOfUse Screen Snapshot', () => {
  it('should render the SettingsTermsOfUse Screen successfully', () => {
    const { toJSON } = renderWithSafeAreaProvider(<SettingsTermsOfUse />)

    expect(toJSON()).toMatchSnapshot()
  })
})
