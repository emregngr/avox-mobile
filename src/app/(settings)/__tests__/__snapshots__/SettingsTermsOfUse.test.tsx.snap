// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SettingsTermsOfUse Screen Snapshot should render the SettingsTermsOfUse Screen successfully 1`] = `
<View
  testID="safe-layout"
>
  <Text
    testID="header-title"
  >
    Terms of Use
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="back-button"
  >
    <Text>
      Back
    </Text>
  </View>
  <RCTScrollView
    contentContainerClassName="px-4"
    contentContainerStyle={
      {
        "paddingBottom": 20,
        "paddingTop": 64,
      }
    }
    indicatorStyle="black"
  >
    <View>
      <Text
        testID="themed-text-body1-text-100"
      >
        Terms of Use
         
1. Terms of Service
         
By using our application, you agree to the following terms. You agree to use this application only for legal purposes and in accordance with these terms.
         
2. User Account and Security
         
When you create an account on our application, you are responsible for protecting the confidentiality of your account information. You are responsible for all activities performed with your account.
         
3. Privacy and Data Use
         
We collect and process your personal data in accordance with our Privacy Policy. By using the application, you consent to the collection and use of your data.
         
4. User Content
         
You are responsible for all content you upload to our application. The content you share must not infringe on copyrights and must comply with the law.
         
5. Prohibited Uses
         
The following actions are strictly prohibited:
- Sharing illegal, harmful, or inappropriate content
- Harming or harassing other users
- Attempting to circumvent the application's security measures
- Using the application for commercial purposes
         
6. Intellectual Property Rights
         
All content, design, and software in our application are the intellectual property of our company. It is prohibited to copy, modify, or distribute this content without permission.
         
7. Service Changes
         
We reserve the right to modify, suspend, or terminate our application and services at any time. We will notify you of such changes.
         
8. Limitation of Liability
         
We provide our application ‘as is’. We do not guarantee that our services will be uninterrupted or error-free. We are not liable for any damages arising from your use of the application.
         
9. Account Termination
         
We reserve the right to suspend or terminate accounts that violate the terms of use. You may terminate your account at any time.
         
10. Contact and Support
         
You can contact our support team with any questions or issues. We will will try to assist you as soon as possible.
         
11. Updates and Changes
         
We may update these terms of use from time to time. We will notify you of any significant changes. Your continued use of the application means that you accept the current terms.
         
12. Applicable Law
         
These terms are governed by and construed in accordance with the laws of the Republic of Turkey. Any disputes arising hereunder shall be resolved by the courts of Turkey.
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
